baseplate/core.py

```diff
@@ -238,6 +244,156 @@ class (means that the authentication payload could not be decrypted
 
         return self.payload.get("sub", None)
 
+    @property
+    def user_roles(self):
+        if not self.defined:
+            raise WithheldAuthenticationError
+
+        roles = self.payload.get("user_roles", None)
+        return set(roles) if roles else set()
+
+    @property
+    def oauth_client_id(self):
+        if not self.defined:
+            raise WithheldAuthenticationError
+
+        return self.payload.get("client_id", None)
+
+    @property
+    def oauth_client_type(self):
```

Comment:
These will probably need docstrings as this object is generated in the sphinx docs.